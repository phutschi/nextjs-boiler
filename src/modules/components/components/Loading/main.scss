@-webkit-keyframes spin {
    0% {
        -webkit-transform: rotate(0deg);
            transform: rotate(0deg);
    }
    100% {
        -webkit-transform: rotate(360deg);
            transform: rotate(360deg);
    }
}

@keyframes spin {
    0% {
        -webkit-transform: rotate(0deg);
            transform: rotate(0deg);
    }
    100% {
        -webkit-transform: rotate(360deg);
            transform: rotate(360deg);
    }
}

.loading {
    border-radius: 50%;
    width: 24px;
    height: 24px;
    border: 0.25rem solid rgba(255, 255, 255, 0.2);
    border-top-color: black;
    -webkit-animation: spin 1s infinite linear;
            animation: spin 1s infinite linear;
}

.loading-wrapper.fullscreen {
    position: fixed;
    width: 100%;
    height: 100%;
    background-color: white;
    z-index: 2;
}

.loading-wrapper.fullscreen .loading {
    position: relative;
    top: 50%;
    left: 50%;
}