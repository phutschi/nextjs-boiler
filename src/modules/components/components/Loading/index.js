/* @flow */

import * as React from 'react';

import './main.scss';

type Props = {
    fullscreen? : boolean
};

const Loading = ( { fullscreen } : Props ) => (
    <div className={`loading-wrapper ${fullscreen ? 'fullscreen' : ''}`}>
        <div className="loading" />
    </div>
);

Loading.defaultProps = { fullscreen : false };

export default Loading;
