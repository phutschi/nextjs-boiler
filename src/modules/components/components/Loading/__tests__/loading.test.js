/* global expect */

import * as React from 'react';

import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import Loading from '../';

describe( '<Loading />', () => {
    it( 'Snapshot', () => {
        const wrapper = renderer.create( <Loading /> ).toJSON();

        expect( wrapper ).toMatchSnapshot();
    } );

    it( 'Renders without props', () => {
        const wrapper = shallow( <Loading /> );

        expect( '' ).toEqual( '' );
    } );
} );
