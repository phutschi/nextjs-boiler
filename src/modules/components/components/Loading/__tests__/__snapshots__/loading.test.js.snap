// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> Snapshot 1`] = `
<div
  className="loading"
/>
`;
